### PR TITLE
feat: Add methods to modify account

### PR DESCRIPTION
**Pull request by rmeissner**
Implements #1889

Adds:
- `evm_setAccountBalance`
- `evm_setAccountCode`
- `evm_setAccountStorageAt`

### CODE REVIEW

**Comment by rmeissner** [comment]
@davidmurdoch I was wondering how the api documentation is generated and if there is a need to manually adjust it (e.g. https://github.com/trufflesuite/ganache/blob/develop/docs/typedoc/api.json)

~Also I would also try to add logic to set the storage and to set the code for an account. Should this happen in different PRs?~
I added all the different rpc methods to this PR for now, let me know if I should split them.

**Comment by davidmurdoch** [comment]
Wow, at first glance this looks very thorough!

The docs are auto-generated as part of the release and end up [here](https://trufflesuite.github.io/ganache/) (for now). Note that these docs are actually just a proof of concept and will likely be rewritten.

**Comment by rmeissner** [comment]
Cool :)

Let me know if I should adjust any of the method names (or anything else). 

I am not sure why some of the tests are failing :/

Also another thing I realized is that `src/packages/ganache/npm-shrinkwrap.json` and `src/chains/filecoin/filecoin/package-lock.json` have changes (I assume from running `npm i`). Not sure if this is ok, else I would revert them.

**Comment by rmeissner** [comment]
@davidmurdoch is there any way to push this forward from my side? I want to use Ganache v7 to simulate Safe transactions and for this I require this functionality :D

**Comment by MicaiahReid** [comment]
@rmeissner I'll review/test this today and we can hopefully get it merged this week!

File: src/chains/ethereum/ethereum/RPC-METHODS.md [hunk 1]
```diff
@@ -851,6 +851,52 @@ Sets the given account's nonce to the specified value. Mines a new block before
 
 ---
 
+#### evm_setAccountBalance
+
+Sets the given account's balance to the specified value. Mines a new block before returning.
```
Comment (MicaiahReid) [inline src/chains/ethereum/ethereum/RPC-METHODS.md]:
```suggestion
Sets the given account's balance to the specified WEI value. Mines a new block before returning.
```

File: src/chains/ethereum/ethereum/RPC-METHODS.md [hunk 2]
```diff
@@ -851,6 +851,52 @@ Sets the given account's nonce to the specified value. Mines a new block before
...
+
+##### Arguments
+
+- `address: DATA` : The account address to update.
+- `balance: QUANTITY` : The balance value to be set.
```
Comment (MicaiahReid) [inline src/chains/ethereum/ethereum/RPC-METHODS.md]:
```suggestion
- `balance: QUANTITY` : The balance value, in WEI, to be set.
```

**Review by davidmurdoch — CHANGES_REQUESTED** [review]
Looks great! Can you drop the filecoin package-lock.json changes? Once that is done this should be good to go on my end!

Excellent work!

**Comment by davidmurdoch** [comment]
Oh, we have a flaky test and a couple of tests that like like to timeout, so don't worry about the tests failures here.

File: src/chains/ethereum/ethereum/src/api.ts [hunk 3]
```diff
@@ -398,6 +398,121 @@ export default class EthereumApi implements Api {
     return true;
   }
 
+  /**
+   * Sets the given account's balance to the specified value. Mines a new block
```
Comment (MicaiahReid) [inline src/chains/ethereum/ethereum/src/api.ts]:
```suggestion
   * Sets the given account's balance to the specified WEI value. Mines a new block
```

File: src/chains/ethereum/ethereum/src/api.ts [hunk 4]
```diff
@@ -398,6 +398,121 @@ export default class EthereumApi implements Api {
...
+   * before returning.
+   *
+   * Warning: this will result in an invalid state tree.
+   *
+   * @param address - The account address to update.
+   * @param balance - The balance value to be set.
```
Comment (MicaiahReid) [inline src/chains/ethereum/ethereum/src/api.ts]:
```suggestion
   * @param balance - The balance value, in WEI, to be set.
```

**Review by MicaiahReid — CHANGES_REQUESTED** [review]
Great work and thanks for the PR! I've requested a few simple changes.

One thing we (@rmeissner, @davidmurdoch, whoever is interested) should probably discuss is data validation for these methods. Currently, if you send bad data for the balance/code/storage, for example `"0xw"`, it will always respond `true`. Also, if there is some valid data, it will update the balance/code/storage _up to_ the bad data. For example, setting balance to `"0xab;lkjf"` will set the balance to `0xab`. This is also true for `evm_setAccountNonce`, not just these new methods.

**Comment by rmeissner** [comment]
Yeah, I followed `evm_setAccountNonce`, but I agree some data validation would be nice. Maybe something for a separate PR, but I could look into this.

**Review by MicaiahReid — APPROVED** [review]

**Comment by davidcallanan** [comment]
Hello, any idea when this is being officially released?

**Pull request merged**